### ISSUE TITLE
`cabal update` aborts everything when _one_ of the configured remotes is offline.

### ISSUE THREAD

**Issue by lspitzner** [haskell/cabal#3382]
- have more than one remote configured
- `cabal update`
- one of the configured remotes answers with 503/504 whatever HTTP status codes
- expected behaviour: the other remotes are still updated properly.
- observed: the process aborts right after printing the error message.

(cabal version is some few-weeks old HEAD (some 1.25.0.0), if it matters)

**Comment by BrianErikson** [comment]
If it helps, I'm getting a `HTTP code 502` error when performing `cabal update`.
```
Downloading the latest package list from hackage.haskell.org
cabal.exe: Failed to download
http://hackage.haskell.org/packages/archive/00-index.tar.gz : ErrorMisc "Error
HTTP code: 502"
```

**Comment by lspitzner** [comment]
@BrianErikson Yes, this ongoing incident is the reason that i discovered this problem. However the described issue exists independent of the incident. I'd really expect `cabal update` to keep working when some of my configured remotes go offline.

(and the admins are aware and investigating, see [status page](https://status.haskell.org/pages/incident/537c07b0cf1fad5830000093/5720f6effe92599160002a31))

**Comment by cartazio** [comment]
mine worked fine ....
